File: service-backbase-java/src/main/java/org/example/backend/model/response/ArticlesListResponseBody.java
package org.example.backend.model.response;

import java.util.List;


public class ArticlesListResponseBody {

  public List<ArticleView> articles;
  public Integer articlesCount;

  public static class ArticleView {

    public String slug;
    public String title;
    public String description;
    public String body;
    public List<String> tagList;
    public String createdAt;
    public String updatedAt;
    public Boolean favorited;
    public Integer favoritesCount;
    public Author author;


    public static class Author {

      public String username;
      public String bio;
      public String image;
      public Boolean following;
    }
  }
}

File: service-backbase-kotlin/src/main/kotlin/org/example/backend/model/response/ArticlesListResponseBody.kt
package org.example.backend.model.response

class ArticlesListResponseBody(
        val articles: List<ArticlesListArticle>? = null,
        val articlesCount: Int? = null,
)

class ArticlesListArticle(
        val slug: String? = null,
        val title: String? = null,
        val description: String? = null,
        val body: String? = null,
        val tagList: List<String>? = null,
        val createdAt: String? = null,
        val updatedAt: String? = null,
        val favorited: Boolean? = null,
        val favoritesCount: Int? = null,
        val author: ArticlesListArticleAuthor? = null,
)

class ArticlesListArticleAuthor(
        val username: String? = null,
        val bio: String? = null,
        val image: String? = null,
        val following: Boolean? = null,
)